src/browser/BrowserCodeReader.ts

```diff
@@ -699,17 +699,15 @@ export class BrowserCodeReader {
         setTimeout(() => loop(), this.timeBetweenScansMillis);
       } catch (e) {
 
+        callbackFn(null, e);
+
         const isChecksumOrFormatError = e instanceof ChecksumException || e instanceof FormatException;
         const isNotFound = e instanceof NotFoundException;
 
-        if (!isChecksumOrFormatError && !isNotFound) {
-          // not expected
-          throw e;
+        if (isChecksumOrFormatError || isNotFound) {
+          // trying again
+          setTimeout(() => loop(), 0);
```

Comment:
What about throwing in the unexpected Exception scenario? I think the loop can't just end silently in that case.

Reply:
Considering the changes I would argue against throwing in this case as it forces the caller to handle the given exception twice.

Other potential solutions:
 1) We can restore the old behavior and use the callback only for expected exception types while throwing anything else.
 2) Remove all of the custom exception handling code and keep the loop alive until the caller decides to stop the scanning process.

I am slightly in favor of the second solution as it would reduce the amount of required exception handling code especially as the caller is already expected to have an exception handling callback in place.